service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentService.java

```diff
@@ -220,6 +230,14 @@
         return updatedC2Bundles;
     }
 
+    public String getUploadedDocumentUserDetails(String authorisation) {
+        UserDetails userDetails = idamClient.getUserDetails(authorisation);
+
+        boolean isHMCTSUser = userDetails.getRoles().stream().anyMatch(UserRole::isHMCTSUser);
+
+        return isHMCTSUser ? "HMCTS" : userDetails.getEmail();
+    }
+
```

Comment:
where should this go? Any existing services it can go (I didn't find any relevant)? or new?

Reply:
`OrganistaitonService` maybe?

Reply:
moved to new class